apps/cow-tools/src/app/milkman/(components)/orderForm/PriceChecker.tsx

```diff
@@ -92,22 +100,297 @@ export function FormSelectPriceChecker({
+          {priceFeeds &&
+            priceFeeds.map((priceFeed, index) => (
+              <Table.BodyRow key={index}>
+                <Table.BodyCell>
+                  {truncateAddress(priceFeed.address)}
+                </Table.BodyCell>
+                <Table.BodyCell>{priceFeed.description}</Table.BodyCell>
+                <Table.BodyCell>
+                  {priceFeed.reversed ? "Yes" : "No"}
+                </Table.BodyCell>
+                <Table.BodyCell>
+                  <div className="flex items-center justify-center gap-x-2">
+                    <button
+                      className="justify-self-center text-amber9 hover:text-amber10 "
+                      type="button"
+                      onClick={() => {
+                        setIndexToEdit(index);
+                        setIsOpen(true);
+                      }}
+                    >
+                      <Pencil1Icon className="w-5 h-5" />
+                    </button>
+                    <button
+                      className="justify-self-center text-tomato9 hover:text-tomato10 "
+                      type="button"
+                      onClick={() => {
+                        setValue("priceFeeds", [
+                          ...priceFeeds.slice(0, index),
+                          ...priceFeeds.slice(index + 1),
+                        ]);
+                      }}
+                    >
+                      <TrashIcon className="w-5 h-5" />
+                    </button>
+                  </div>
+                </Table.BodyCell>
+              </Table.BodyRow>
+            ))}
+        </Table.Body>
+      </Table>
+      {errors.priceFeeds && (
+        <FormMessage className="h-6 text-sm text-tomato10 w-full">
+          <span>{errors.priceFeeds.message as string}</span>
+        </FormMessage>
+      )}
+    </div>
+  );
+}
+
+function ChainlinkAddModal({
+  priceCheckerForm,
+  setIsOpen,
+  indexToEdit = -1,
+}: {
+  priceCheckerForm: UseFormReturn;
+  setIsOpen: (open: boolean) => void;
+  indexToEdit?: number;
+}) {
+  const {
+    setValue: setValuePriceCheckerForm,
+    getValues: getValuesPriceCheckerForm,
+  } = priceCheckerForm;
+
+  const publicClient = createPublicClient({
+    chain: goerli,
+    transport: http(),
+  });
+  const priceFeedChainlinkSchema = getPriceFeedChainlinkSchema(publicClient);
+
+  useEffect(() => {
+    if (indexToEdit !== -1) {
+      const priceFeeds = getValuesPriceCheckerForm("priceFeeds") as {
+        address: Address;
+        description: string;
+        reversed: boolean;
+      }[];
+      const priceFeed = priceFeeds[indexToEdit];
+      setValue("priceFeedAddress", priceFeed.address);
+      setValue("reversed", priceFeed.reversed);
+    }
+  }, []);
+
+  const newPriceFeedForm = useForm<typeof priceFeedChainlinkSchema._type>({
+    resolver: zodResolver(priceFeedChainlinkSchema),
+  });
+  const { register, watch, setValue } = newPriceFeedForm;
+  const formData = watch();
```

Comment:
couldn't we just use the parent form instead of adding another one here?

Reply:
I divided it into two forms because it would be easier to test if each price feed is the contract that we expect here. Otherwise, we would have to deal with some errors on the other component. Maybe we can take a look at it together later, what do you think about it?